modules/meilisearch/meilisearch.go

```diff
@@ -0,0 +1,119 @@
+package meilisearch
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultMasterKey = "just-a-master-key-for-test"
+	defaultHTTPPort  = "7700/tcp"
+	masterKeyEnvVar  = "MEILI_MASTER_KEY"
+)
+
+// MeilisearchContainer represents the Meilisearch container type used in the module
+type MeilisearchContainer struct {
+	testcontainers.Container
+	MasterKey string
+}
+
+// Run creates an instance of the Meilisearch container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*MeilisearchContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{defaultHTTPPort},
+		Env: map[string]string{
+			masterKeyEnvVar: defaultMasterKey,
+		},
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	// Gather all config options (defaults and then apply provided options)
+	settings := defaultOptions()
+	for _, opt := range opts {
+		if apply, ok := opt.(Option); ok {
+			apply(settings)
+		}
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, fmt.Errorf("customize: %w", err)
+		}
+	}
+
+	if settings.DumpDataFilePath != "" {
+		genericContainerReq.Files = []testcontainers.ContainerFile{
+			{
+				HostFilePath:      settings.DumpDataFilePath,
+				ContainerFilePath: "/dumps/" + settings.DumpDataFileName,
+				FileMode:          0o755,
+			},
+		}
+		genericContainerReq.Cmd = []string{"meilisearch", "--import-dump", fmt.Sprintf("/dumps/%s", settings.DumpDataFileName)}
+	}
+
+	// the wait strategy does not support TLS at the moment,
+	// so we need to disable it in the strategy for now.
+	genericContainerReq.WaitingFor = wait.ForHTTP("/health").
+		WithPort(defaultHTTPPort).
+		WithTLS(false).
+		WithStartupTimeout(120 * time.Second).
+		WithStatusCodeMatcher(func(status int) bool {
+			return status == 200
```

Comment:
suggestion: use the http constant.
```suggestion
			return status == http.StatusOK
```